source/MaterialXRenderGlsl/GlslValidator.h

```diff
@@ -152,6 +162,9 @@ class GlslValidator : public ShaderValidator
     /// Flag to indicate if validator has been initialized properly.
     bool _initialized;
 
+    /// Data type being rendered out to texture
+    std::string _type = "color3";
```

Comment:
This data should be knowable from the input type string (or better yet, InputPtr) that was stored in the GenOptions instance by the client.